applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -126,6 +140,151 @@ enum {
 
 static K_EVENT_DEFINE(usb_event);
 
+static void report_sent(struct usb_hid_device *usb_hid, struct usb_hid_buf *buf, bool error);
+
+static uint8_t usb_hid_buf_get_report_id(struct usb_hid_buf *buf)
+{
+	__ASSERT_NO_MSG(buf->status_bm & USB_HID_BUF_ALLOCATED);
+	uint8_t report_id = REPORT_ID_COUNT;
+
+	if (!(buf->status_bm & USB_HID_BUF_BOOT_REPORT_FORMAT)) {
+		report_id = buf->data[0];
+	} else {
+		if (IS_ENABLED(CONFIG_DESKTOP_HID_BOOT_INTERFACE_MOUSE)) {
+			report_id = REPORT_ID_BOOT_MOUSE;
+		} else if (IS_ENABLED(CONFIG_DESKTOP_HID_BOOT_INTERFACE_KEYBOARD)) {
+			report_id = REPORT_ID_BOOT_KEYBOARD;
+		}
+	}
+
+	/* Should not happen. */
+	__ASSERT_NO_MSG(report_id != REPORT_ID_COUNT);
+
+	return report_id;
+}
+
+static struct usb_hid_buf *usb_hid_buf_alloc(struct usb_hid_device *usb_hid, const uint8_t *data,
+					     size_t size)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(usb_hid->report_bufs); i++) {
+		struct usb_hid_buf *r = &usb_hid->report_bufs[i];
+
+		if (!(r->status_bm & USB_HID_BUF_ALLOCATED)) {
+			__ASSERT_NO_MSG(!(r->status_bm & USB_HID_BUF_SENDING));
+			__ASSERT_NO_MSG(!(r->status_bm & USB_HID_BUF_BOOT_REPORT_FORMAT));
```

Comment:
you can overall check if the status_bm is zero as this is how you free the buffer

Reply:
We can desist from extra checks here and simplify